concent_api/core/views.py

```diff
@@ -1443,6 +1443,114 @@ def store_message(
+    else:
+        if len(clients_subtask_list_as_requestor) != 0:
+            update_states_of_subtasks(clients_subtask_list_as_requestor)
+            logging.log_changes_in_subtask_states(
+                client_public_key,
+                len(clients_subtask_list_as_requestor),
+                'requestor',
+            )
+        if len(clients_subtask_list_as_provider) != 0:
+            update_states_of_subtasks(clients_subtask_list_as_provider)
+            logging.log_changes_in_subtask_states(
+                client_public_key,
+                len(clients_subtask_list_as_provider),
+                'provider',
+            )
+
+def update_states_of_subtasks(
+    subtask_list:       list,
+    client_public_key:  str,
+):
+    current_time = get_current_utc_timestamp()
+
+    for subtask in subtask_list:
+        receive_queue_item = ReceiveQueue.objects.get(
+            client  = client_public_key,
+            subtask = subtask,
+        )
+        if subtask.next_deadline.timestamp() <= current_time:
+            if subtask.state == Subtask.SubtaskState.FORCING_REPORT:
+                subtask.state = Subtask.SubtaskState.REPORTED
+                subtask.full_clean()
+                subtask.save()
+                create_or_update_receive_queue(
+                    response_type   = ReceiveQueue.ResponseType.ForceReportComputedTask,
+                    client          = client_public_key,
+                    queue           = receive_queue_item.queue,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            elif subtask.state == Subtask.SubtaskState.FORCING_RESULT_TRANSFER:
+                subtask.state = Subtask.SubtaskState.RESULT_UPLOADED
+                subtask.full_clean()
+                subtask.save()
+                create_or_update_receive_queue(
+                    response_type   = ReceiveQueue.ResponseType.ForceGetTaskResultUpload,
+                    client          = client_public_key,
+                    queue           = receive_queue_item.queue,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            elif subtask.state == Subtask.SubtaskState.FORCING_ACCEPTANCE:
+                subtask.state = Subtask.SubtaskState.REJECTED
+                subtask.full_clean()
+                subtask.save()
+                create_or_update_receive_queue(
+                    response_type   = ReceiveQueue.ResponseType.ForceSubtaskResultsResponse,
+                    client          = client_public_key,
+                    queue           = receive_queue_item.queue,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            # TODO when last use case will be implemented
+            # elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION:
+
+
+def store_or_update_receive_queue(
```

Comment:
I don't think that object in ReceiveQueue should be ever updated. If it is pass deadline, it should be retrieved by client too, and here you should only create new entry. @cameel what do you think ?